tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,190 @@
+        )
+
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        # Stop writing data to the image by killing the FIO process,
+        # then start FIO again and check cache status
+        log.info("Test to kill FIO process and restarting FIO to check cache status")
+        with parallel() as p:
+            p.spawn(run_fio, **fio_ready(config, cache_client))
+            p.spawn(kill_fio, pwl.rbd)
+
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        # Restart mon service along with cache test
+        log.info("Test to restarting mon service along with pwl cache test")
+        with parallel() as p:
+            p.spawn(rados_obj.restart_daemon_services, "mon")
+            verify_cache(pwl, config, cache_client, image_spec)
+
+        # Restart osd service along with cache test
+        log.info("Test to restarting osd service along with pwl cache test")
+        cmd = "ceph orch ls osd"
+        out = cache_client.exec_command(cmd=cmd, output=True)
+
+        # Use regular expression to extract the first OSD name
+        match = re.search(r"osd\.(\S+)", out)
+
+        with parallel() as p:
+            p.spawn(rados_obj.restart_daemon_services, match.group(0))
+            verify_cache(pwl, config, cache_client, image_spec)
+
+        # Restart the ceph cluster along with cache test
+        log.info("Test to  restarting ceph target service along with pwl cache test")
+        cmd = "systemctl restart ceph.target"
+        admin_node.exec_command(cmd=cmd, sudo=True)
+
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        mon_host = kw["ceph_cluster"].get_nodes(role="mon")[0]
+        # Test to remove mon service
+        log.info("Test to remove mon service along with pwl cache test")
+        if not mon_obj.remove_mon_service(host=mon_host.hostname):
+            log.error("Could not remove the mon host")
+            raise Exception("mon service not removed error")
+
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        # Adding the mon back to the cluster
+        log.info("Test to add mon service back along with pwl cache test")
+        if not mon_obj.add_mon_service(host=mon_host):
+            log.error("Could not add mon service")
+            raise Exception("mon service not added error")
+
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        # Test to remove osd service
+        log.info("Test to remove osd service along with pwl cache test")
+        pg_set = rados_obj.get_pg_acting_set(pool_name=pool)
+        target_osd = pg_set[0]
+        utils.osd_remove(ceph_cluster, target_osd)
+
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        # Test to add the removed OSD back and checking the cluster status
+        log.info("Test to add osd service back along with pwl cache test")
+        utils.add_osd(ceph_cluster, host.hostname, dev_path, target_osd)
+
+        verify_cache(pwl, config, cache_client, image_spec)
```

Comment:
Please implement osd scrubbing as well as mentioned in the test steps.

Reply:
Done.